Allow for flexible logo sizes

ryelle opened the pull request:
A larger image can be uploaded and displayed, and the CSS will take care of making it the expected size on the frontend. Using this method, we can remove the `retina_logo` option, because larger image sizes will be supported.

See #57 

To test:

- Open the customizer and choose a large (1000px+) image for your site logo
- You should be able to skip the cropping step, or crop it to the 3:1 aspect ratio
- When saved, you see the image in the header & footer 👍

ryelle committed:
Allow for flexible logo sizes

A larger image can be uploaded and displayed, and the CSS will take care of making it the expected size on the frontend

melchoyce commented:
Just tested this. Header works great! I'm seeing the logo at full size in the footer, though:

![image](https://user-images.githubusercontent.com/2846578/94952277-b6f37e00-04b3-11eb-8ded-a8bf72a9077c.png)

LittleBigThing commented:
I used an image with a different aspect ration (than 3:1) and it does not seem to look as expected when the logo is set without resizing the image:

![Screenshot 2020-10-02 at 21 32 41](https://user-images.githubusercontent.com/5901923/94962600-01e2b500-04f7-11eb-96cd-0e5858ca9e28.png)

It is due to the max-width and max-height defined [here](https://github.com/WordPress/twentytwentyone/blob/118e3d585da225fe600d2ee6c77bc955a62e42cf/assets/sass/06-components/header.scss#L88), based on the fixed width and height.

We should either set a `max-height` or a `max-width` and set the other to auto?
In Twenty Twenty, a `max-height` is set and the `width` is auto, this is probably the way to go except when the logo can be centered and then it could be higher? But that is going to make things more complicated.

And indeed, the full-size image visible in the footer.

ryelle committed:
Wrap the footer logo in the same container as the header

ryelle committed:
Set the width to auto to prevent skewed logo

ryelle commented:
I've added the same wrapper to the footer logo as is in the header, so now it's contained using the same CSS. I've also added `width:auto` so the image isn't skewed when it's not the right aspect ratio - this way the image should always be in the 300x100 container, but will scale down rather than skew.

jffng reviewed (approved):
I like this approach.

Could be addressed in a separate PR but should the `.site-logo` margins be adjusted so the logo is left-aligned with the site title?

![Screen Shot 2020-10-05 at 10 16 27 AM](https://user-images.githubusercontent.com/5375500/95111116-e2fd4200-06f3-11eb-8e5e-0b6fd7b6d4d8.png)

melchoyce commented:
Margins fixed in #246.
